File: backend/app/api/schemas/alerts.py
"""Contratos de alerta de anomalia."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.core.enums import AlertSeverity, AlertStatus


class ClinicalAlertRead(BaseModel):
    id: uuid.UUID
    patient_id: uuid.UUID
    observation_id: uuid.UUID | None
    signal_key: str
    severity: AlertSeverity
    status: AlertStatus
    detector_source: str
    confidence: float | None
    evidence: dict
    expected_action: str
    detected_at: datetime

    acknowledged_by: str | None
    acknowledged_at: datetime | None
    escalated_to: str | None
    escalated_at: datetime | None
    escalation_reason: str | None
    resolved_by: str | None
    resolved_at: datetime | None
    resolution_notes: str | None

    created_at: datetime

    model_config = {"from_attributes": True}


class AlertSeverityCounts(BaseModel):
    """Quantidade de alertas por severidade para um paciente, usada nos
    "big numbers" do painel de alertas da tela de paciente. Conta todos
    os status (aberto, reconhecido, escalado, encerrado); o filtro por
    status continua disponivel na listagem detalhada apos escolher uma
    severidade."""

    critical: int = 0
    high: int = 0
    moderate: int = 0


class EscalateAlertRequest(BaseModel):
    escalated_to: str = Field(..., min_length=1, max_length=200)
    reason: str = Field(..., min_length=10, max_length=2000)


class ResolveAlertRequest(BaseModel):
    notes: str = Field(..., min_length=10, max_length=2000)

File: frontend/src/types/alerts.ts
/**
 * Tipos de alerta de anomalia.
 * Espelha backend/app/api/schemas/alerts.py.
 */
import type { AlertSeverity, AlertStatus } from "./enums.generated";

export interface ClinicalAlert {
  id: string;
  patient_id: string;
  observation_id: string | null;
  signal_key: string;
  severity: AlertSeverity;
  status: AlertStatus;
  detector_source: string;
  confidence: number | null;
  evidence: Record<string, unknown>;
  expected_action: string;
  detected_at: string;
  acknowledged_by: string | null;
  acknowledged_at: string | null;
  escalated_to: string | null;
  escalated_at: string | null;
  escalation_reason: string | null;
  resolved_by: string | null;
  resolved_at: string | null;
  resolution_notes: string | null;
  created_at: string;
}

export interface EscalateAlertInput {
  escalated_to: string;
  reason: string;
}

export interface ResolveAlertInput {
  notes: string;
}

/** Contagem de alertas por severidade (todos os status), usada nos "big
 * numbers" do painel de alertas antes de escolher uma severidade para
 * ver o detalhe paginado. */
export interface AlertSeverityCounts {
  critical: number;
  high: number;
  moderate: number;
}
